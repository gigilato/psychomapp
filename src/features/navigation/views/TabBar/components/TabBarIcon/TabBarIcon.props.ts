import { IPressableProps, IconName } from '$atoms'

export interface ITabBarIconProps extends Omit<IPressableProps, 'ID'> {
  isSelected: boolean
  icon: IconName
  route: string
  name: string
}
