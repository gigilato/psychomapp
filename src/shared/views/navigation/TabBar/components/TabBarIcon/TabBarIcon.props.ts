import { Route } from 'expo-router'

import { IPressableProps, IconName } from '$atoms'

export interface ITabBarIconProps extends Omit<IPressableProps, 'ID'> {
  isSelected: boolean
  icon: IconName
  route: Route<string>
  name: string
}
