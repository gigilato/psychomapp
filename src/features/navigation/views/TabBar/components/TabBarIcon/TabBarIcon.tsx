import clsx from 'clsx'
import { router } from 'expo-router'

import { Icon, Pressable, Text } from '$atoms'
import { impactAsync } from '$libs/haptics'

import { ITabBarIconProps } from './TabBarIcon.props'

export const TabBarIcon = ({ route, icon, isSelected, name }: ITabBarIconProps) => {
  const color = isSelected ? 'text-primary-classic' : ''
  return (
    <Pressable
      ID={`TabBarIcon/${name}`}
      className="items-center w-[90]"
      onPress={() => {
        impactAsync()
        router.push(route)
      }}
    >
      <Icon name={icon} size={30} className={clsx(color)} />
      <Text className={clsx(color)}>{name}</Text>
    </Pressable>
  )
}
