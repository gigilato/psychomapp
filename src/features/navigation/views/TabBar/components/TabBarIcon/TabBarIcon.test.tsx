import { render } from '@testing-library/react-native'

import { TabBarIcon } from './TabBarIcon'

describe('TabBarIcon', () => {
  it('should render selected', () => {
    const component = render(<TabBarIcon icon="bank" isSelected name="icon" route="/dashboard/" />)
    expect(component).toMatchSnapshot()
  })

  it('should render selected', () => {
    const component = render(
      <TabBarIcon icon="bank" isSelected={false} name="icon" route="/dashboard/profile/" />
    )
    expect(component).toMatchSnapshot()
  })
})
