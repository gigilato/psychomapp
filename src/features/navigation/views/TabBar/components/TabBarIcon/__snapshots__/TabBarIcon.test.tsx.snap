// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon should render selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "width": 90,
      },
      [
        {
          "opacity": undefined,
        },
        undefined,
      ],
    ]
  }
  testID="TabBarIcon/icon"
>
  <Text />
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "color": "#FFB300",
        },
      ]
    }
  >
    icon
  </Text>
</View>
`;

exports[`TabBarIcon should render selected 2`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "width": 90,
      },
      [
        {
          "opacity": undefined,
        },
        undefined,
      ],
    ]
  }
  testID="TabBarIcon/icon"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 30,
        },
        undefined,
        {
          "fontFamily": "IcoMoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ?
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
      ]
    }
  >
    icon
  </Text>
</View>
`;
